apps/chat-e2e/src/tests/ChatSwitch.test.ts

```diff
@@ -0,0 +1,160 @@
+      expect
+        .soft(style[0], `Conversation cursor is ${expectedCursor}`)
+        .toBe(expectedCursor);
+    };
+
+    const verifySelectedConversation = async (conversationName: string) => {
+      await expect
+        .soft(
+          conversations.selectedConversation(conversationName),
+          ExpectedMessages.conversationIsSelected,
+        )
+        .toBeVisible();
+    };
+
+    await dialTest.step('Prepare two empty conversations', async () => {
+      firstConversation = conversationData.prepareEmptyConversation();
+      conversationData.resetData();
+      secondConversation = conversationData.prepareEmptyConversation();
+      await dataInjector.createConversations([
+        firstConversation,
+        secondConversation,
+      ]);
+      await localStorageManager.setSelectedConversation(firstConversation);
+    });
+
+    await dialTest.step('Send request to the first conversation', async () => {
+      await dialHomePage.openHomePage();
+      await dialHomePage.waitForPageLoaded();
+      await dialHomePage.throttleAPIResponse(
+        API.chatHost,
+        responseThrottlingTimeout,
+      );
+      await chat.sendRequestWithButton(request, false);
+      firstConversation.name = request;
+    });
+
+    await dialTest.step(
+      'Verify any conversation cursor is "not-allowed" during the text generation',
+      async () => {
+        await verifyConversationCursor(
+          firstConversation.name,
+          Cursors.notAllowed,
+        );
+        await verifyConversationCursor(
+          secondConversation.name,
+          Cursors.notAllowed,
+        );
+      },
+    );
+
+    await dialTest.step(
+      'Verify another conversation is not selectable during text generation',
+      async () => {
+        await conversations.getEntityByName(secondConversation.name).click();
+        await verifySelectedConversation(firstConversation.name);
+      },
+    );
+
+    await dialTest.step(
+      'Select "Replay" option from the first conversation menu',
+      async () => {
+        await chatMessages.waitForResponseReceived();
+        await conversations.openEntityDropdownMenu(firstConversation.name);
+        await conversations.selectEntityMenuOption(MenuOptions.replay, {
+          triggeredHttpMethod: 'POST',
+        });
+        thirdConversationName = `${ExpectedConstants.replayConversation}${firstConversation.name}`;
+        await chat.startReplay();
+      },
+    );
+
+    await dialTest.step(
+      'Verify conversation cursor is "not-allowed" during the chat replay',
+      async () => {
+        await verifyConversationCursor(
+          secondConversation.name,
+          Cursors.notAllowed,
+        );
+      },
+    );
+
+    await dialTest.step(
+      'Open compare mode for replayed conversation',
+      async () => {
+        await chatMessages.waitForResponseReceived();
+        await conversations.openEntityDropdownMenu(firstConversation.name, 2);
```

Comment:
there is a way to open conversations in compare mode if set them selected before app launch:

`await localStorageManager.setSelectedConversation(firstConversation,secondConversation);`

You can rearrange the test to be first in the chain preventing unnecessary UI actions

Reply:
done